pytorch_lightning/core/memory.py

```diff
@@ -438,3 +445,17 @@ def get_human_readable_count(number: int) -> str:
         return f"{int(number):,d} {labels[index]}"
 
     return f"{number:,.1f} {labels[index]}"
+
+
+def _is_lazy_weight_tensor(p: Tensor) -> bool:
+    if _TORCH_GREATER_EQUAL_1_8:
+        from torch.nn.parameter import UninitializedParameter
+        if isinstance(p, UninitializedParameter):
```

Comment:
I would actually move this check inside the `trainable_parameters` and `trainable_parameters` property.

 `_is_lazy_weight_tensor` is just a function detect if it is an  UninitializedParameter parameter.

Reply:
Are you talking about the `_TORCH_GREATER_EQUAL_1_8` check?

Reply:
I meant this:

```

    @property
    def num_parameters(self) -> int:
        """ Returns the number of parameters in this module. """
        num_parameters = 0
        for p in self._module.parameters()
            if not _is_lazy_weight_tensor(p):
                num_parameters += np.prod(p.shape)
            else:
                    # some parameters aren't initialised yet.
                    warning_cache.warn(
                        "A layer with UninitializedParameter was found. "
                        "Thus, the total number of parameters detected may be inaccurate."
                    )
        return num_parameters

def _is_lazy_weight_tensor(p: Tensor) -> bool:
    if _TORCH_GREATER_EQUAL_1_8:
        from torch.nn.parameter import UninitializedParameter
        return isinstance(p, UninitializedParameter):
    return False
```

Reply:
I personally prefer the current approach considering `_is_lazy_weight_tensor` is not used elsewhere

Reply:
If we can get a consensus on this, I can make the change promptly.

Reply:
Ok, let stick with this approach then.

Reply:
Okay sure.